### ISSUE TITLE
Gatsby site freezes after swiping to go back in iOS Safari

### ISSUE THREAD

**Issue by slk333** [gatsbyjs/gatsby#11355]
see: https://github.com/gatsbyjs/gatsby/issues/2842
please re-open the issue

Currently on every Gatsby site, when swiping to go back in mobile safari, the screen freezes up and sometimes goes blank for ~2-6 seconds. This really breaks the flow of the site.

just try on gatsby website itself

gatsbyfreeze

As soon as the page go backs I try to scroll, but its completely frozen (until that highlighted state on the button goes away).

**Comment by sidharthachatterjee** [comment]
I can't seem to reproduce this on http://gatsbyjs.org on my iPhone X running iOS 12.1.3

As far as I know, this was fixed in https://github.com/taion/scroll-behavior/pull/129

**Comment by sidharthachatterjee** [comment]
Can also confirm that `scrollRestoration` is set to `auto`
![screenshot 2019-01-28 22 17 31](https://user-images.githubusercontent.com/7701981/51851790-9bcd8400-234a-11e9-81dc-4ed263f74409.png)

**Comment by slk333** [comment]
from the tests I've made, this freeze only appear on iOS 12.2 (I've tested iOS 12.1 and 12.2 simulators, problem only appears on iOS 12.2)
Thank you for your feedback @sidharthachatterjee 
What could I do from here?
Is the problem only attributed to Apple?

**Comment by slk333** [comment]
https://youtu.be/Gz3cGhwnHZM

**Comment by sidharthachatterjee** [comment]
The issue at https://github.com/gatsbyjs/gatsby/issues/2842 was an iOS Safari bug so yes, looks like this is specific to iOS. 

Haven't tested this on the iOS 12.2 beta yet.

**Comment by rutzfutz** [comment]
Can you please send us the output of this site using the browser that you experience the freezing issue on?

https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_nav_all

It just outputs all navigator object properties. This may help as the freezing fix unfortunately had to rely on browser detection, which uses some of these properties.

**Comment by slk333** [comment]
Hi and thanks for the reply.

Browser CodeName: Mozilla

Browser Name: Netscape

Browser Version: 5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/71.0.3578.89 Mobile/15E148 Safari/605.1

Cookies Enabled: true

Browser Language: en-US

Browser Online: true

Platform: iPhone

User-agent header: Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/71.0.3578.89 Mobile/15E148 Safari/605.1

**Comment by rutzfutz** [comment]
Thank you. Your browser is still identified correctly by the detection we added in https://github.com/taion/scroll-behavior/pull/129, so this part of the scroll-behavior package is unlikely to cause the freeze you're encountering, as it's not setting `window.history.scrollRestoration` in your scenario.

The bug must lie somewhere else. Sorry not to be of more help!

**Comment by slk333** [comment]
Yes, hopefully this will be fixed by Apple before iOS 12.2 ships. I don't know if we should keep the issue open in the mean time. I can open a new one if ever the bug still remains when iOS 12.2 ships (out of beta)

**Comment by sidharthachatterjee** [comment]
Let's close the issue for now @slk333 since it affects beta iOS. Please feel free to reopen on release if this still occurs. Thank you!
